src/directives/leaflet.js

```diff
@@ -39,6 +39,7 @@ angular.module("leaflet-directive", []).directive('leaflet',
                 genDispatchMapEvent = leafletEvents.genDispatchMapEvent,
                 mapEvents = leafletEvents.getAvailableMapEvents();
 
+            scope.mapId =  attrs.id;
```

Comment:
Have someway of identifying what map an event came from. This is because map events are broadcasted for all maps.